pandas/io/excel/_xlrd.py

```diff
@@ -95,9 +105,30 @@ def _parse_cell(cell_contents, cell_typ):
                     cell_contents = val
             return cell_contents
 
-        data = []
+        data: List[List[Scalar]] = []
 
+        if nrows is not None:
+            _validate_integer("nrows", nrows)
+        header = 0 if header is None else header
+        skiprows = 0 if skiprows is None else skiprows
+        if isinstance(header, list) or isinstance(skiprows, list):
+            nrows = None
         for i in range(sheet.nrows):
+
+            if nrows is not None:
```

Comment:
This block of code looks repeated in the various excel files. Could we turn it into a shared function?

Reply:
Opened excel file is won't be closed if exception associated with the value of nrows is raised in `get_sheet_data` methods, so I moved `_validate_integer` to `read_excel` method in `_base.py` file.

Reply:
Right. But we could create another function in `_base.py` that generates flags to `continue` or `break` bases on this repeated code. 

Same for the block you added above

```
        header = 0 if header is None else header
        skiprows = 0 if skiprows is None else skiprows
        if isinstance(header, list) or isinstance(skiprows, list):
            nrows = None
```